### PR TITLE
Add appi v2 ework

### PR DESCRIPTION
**Pull request by bastikk**
Following endpoints were covered: 
1. [Get User Info](https://github.com/test-IO/cirro-ruby-client/tree/feature/add_api_v2_support#get-user-info) GET /users/:id
2. [Create a Gig](https://github.com/test-IO/cirro-ruby-client/tree/feature/add_api_v2_support#create-a-gig) POST /gigs
3. [Get List of GigInvitations](https://github.com/test-IO/cirro-ruby-client/tree/feature/add_api_v2_support#get-list-of-gig-invitations) GET /gig_invitations
4. [Accept a GigInvitation](https://github.com/test-IO/cirro-ruby-client/tree/feature/add_api_v2_support#accept-a-gig-invitation) POST /gig_invitations/:id/accept

Tested with specs

### CODE REVIEW

File: lib/cirro_io_v2/request_clients/jwt.rb [hunk 1]
```diff
@@ -18,7 +18,8 @@ def initialize(base_url:, private_key:, client_id:)
         end
       end
 
-      def make_request(http_method, url, body = nil, params = nil, _headers = {})
+      def make_request(http_method, url, body: nil, params: nil, _headers: {})
+        # TODO: why do you use underscore here?
```
Comment (thebluber) [inline lib/cirro_io_v2/request_clients/jwt.rb:22]: You mean `_headers`? it must have come from the linter. It's not used in the method.
I suppose it should be added to the request headers:

```
request.headers = headers.merge('Authorization' => bearer_token)
```

File: lib/cirro_io_v2/resources/gig.rb [hunk 2]
```diff
@@ -0,0 +1,18 @@
+module CirroIOV2
+  module Resources
+    class Gig < Base
+      def create(title:, description:, url:, start_at:, end_at:, total_seats:, invitation_mode:, filter_query:,
```
Comment (thebluber) [inline lib/cirro_io_v2/resources/gig.rb:4]: how about passing in `params = {}`? it's more flexible.
there's no need to validate anything in the gem the API will take care of validating params anyway.

File: lib/cirro_io_v2/resources/gig_ivitation.rb [hunk 3]
```diff
@@ -0,0 +1,18 @@
+module CirroIOV2
+  module Resources
+    class GigInvitation < Base
+      def list(user_id: nil, gig_id: nil, before: nil, after: nil, status: nil, limit: 10)
```
Comment (thebluber) [inline lib/cirro_io_v2/resources/gig_ivitation.rb:4]: here the same with `query_params = {}`

If you want to validate you could add a constant listing allowed param keys e.g.:

```
ALLOWED_PARAMS = [:user_id, :gig_id, :limit, :before, :after, :status]

raise ParamNotAllowed if (query_params.keys - ALLOWED_PARAMS).any?
```

File: lib/cirro_io_v2/resources/gig.rb [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+                 tasks:, notification_payload: nil, epam_options: nil)
+        # TODO: Is it ok? as rubocop says:  Avoid parameter lists longer than 5 parameters.
+        # [6/5] (https://rubystyle.guide#too-many-params)
+
+        body = { title: title, description: description, url: url, start_at: start_at, end_at: end_at,
+                 total_seats: total_seats, invitation_mode: invitation_mode, filter_query: filter_query, tasks: tasks,
+                 notification_payload: notification_payload, epam_options: epam_options }
+        # TODO: should I have some validations for body params?
+        response = client.request_client.request(:post, resource_root, body: body)
+        OpenStruct.new(response.body)
```
Comment (thebluber) [inline lib/cirro_io_v2/resources/gig.rb:14]: I was actually hoping to have our own response classes e.g. Responses::Gig which can be a struct of course.
Also please don't use OpenStruct, if possible always use Struct instead.
https://www.rubydoc.info/gems/rubocop/0.61.1/RuboCop/Cop/Performance/OpenStruct

File: lib/cirro_io_v2/resources/responses/gig_invitation_responce.rb [hunk 5]
```diff
@@ -0,0 +1,27 @@
+module CirroIOV2
+  module Resources
+    module Responses
+      class GigInvitationResponce
```
Comment (thebluber) [inline lib/cirro_io_v2/resources/responses/gig_invitation_responce.rb:4]: typo: Response

File: lib/cirro_io_v2/resources/responses/gig_invitation_responce.rb [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+      class GigInvitationListResponce
+        GIG_INVITATION_PARAMS = %i[object url has_more data].freeze
+
+        def self.create(body)
+          gig = Struct.new(*GIG_INVITATION_PARAMS)
+          gig.new(
+            *body.slice(*GIG_INVITATION_PARAMS.excluding(:data)).values,
+            body[:data].map { |gig_invitation| GigInvitationResponce.create(gig_invitation) },
+          )
+          #  TODO: Should I do in this way?
```
Comment (thebluber) [inline lib/cirro_io_v2/resources/responses/gig_invitation_responce.rb:22]: list has attributes like has_more which is important for pagination. Therefore it should be a response object as well, with data as array of GigInvitationResponse objects.

File: lib/cirro_io_v2/resources/responses/user_responce.rb [hunk 7]
```diff
@@ -0,0 +1,14 @@
+module CirroIOV2
+  module Resources
+    module Responses
+      class UserResponce
+        USER_PARAMS = %i[id object first_name last_name time_zone screen_name country_code epam worker].freeze
+
+        def self.create(body)
+          gig = Struct.new(*USER_PARAMS)
```
Comment (thebluber) [inline lib/cirro_io_v2/resources/responses/user_responce.rb:8]: not all structs are gigs... this one is hopefully a user.

File: lib/cirro_io_v2/resources/responses/gig_invitation_responce.rb [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+          gig = Struct.new(*GIG_INVITATION_PARAMS)
+          gig.new(*body.slice(*GIG_INVITATION_PARAMS).values)
+        end
+      end
+
+      class GigInvitationListResponce
+        GIG_INVITATION_PARAMS = %i[object url has_more data].freeze
+
+        def self.create(body)
+          gig = Struct.new(*GIG_INVITATION_PARAMS)
```
Comment (thebluber) [inline lib/cirro_io_v2/resources/responses/gig_invitation_responce.rb:17]: not a gig

File: lib/cirro_io_v2/resources/responses/gig_responce.rb [hunk 9]
```diff
@@ -0,0 +1,15 @@
+module CirroIOV2
+  module Resources
+    module Responses
+      class GigResponce
+        GIG_PARAMS = %i[id object title description url start_at end_at total_seats invitation_mode
+                        filter_query tasks notification_payload epam_options].freeze
+
+        def self.create(body)
```
Comment (thebluber) [inline lib/cirro_io_v2/resources/responses/gig_responce.rb:8]: can we maybe change `create` to `build` or `new`?
it is not creating anything it is really just a wrapper to the server response

**Review by thebluber — CHANGES_REQUESTED** [review]
2 blockers for me:
1. Typo: response
2. Can we please put response objects into their own folder, which should not be nested under resources/

File: lib/cirro_io_v2/resources/responses/gig_responce.rb (hunk 9 above)
Comment (bastikk) [reply lib/cirro_io_v2/resources/responses/gig_responce.rb:8]: Then maybe wrap?

File: lib/cirro_io_v2/resources/responses/gig_invitation_responce.rb (hunk 5 above)
Comment (bastikk) [reply lib/cirro_io_v2/resources/responses/gig_invitation_responce.rb:4]: Thanks, fixed.

File: lib/cirro_io_v2/resources/responses/gig_invitation_responce.rb (hunk 8 above)
Comment (bastikk) [reply lib/cirro_io_v2/resources/responses/gig_invitation_responce.rb]: Thanks, fixed

File: lib/cirro_io_v2/resources/responses/gig_responce.rb (hunk 9 above)
Comment (bastikk) [reply lib/cirro_io_v2/resources/responses/gig_responce.rb]: Good point, I've renamed it to build

**Review by thebluber — APPROVED** [review]
looks good to me now 👍

**Pull request merged**